hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```
@@ -240,7 +300,16 @@ extension (self: Blk)
     Blk(self.stats, f(self.res))
 
 
-sealed trait Statement extends AutoLocated with ProductWithExtraInfo:
+sealed trait Statement extends AutoLocated, ProductWithExtraInfo:
+  
+  override def clone: Statement = this match
+    case t: Term => t.clone
+    case d: Definition => d // Can we clone `Definition`s?
```

Of course, why not? The real question is: _do we need to_? Can the pattern compiler end up duplicating a definition?

If not, just make this `???` for now. I dont know why I need to ask you again and again to prefer crashing rather than silently do the wrong thing.

---

I don't think we need to clone them. I marked this case as `???`.